cbn_cli.py

```diff
@@ -0,0 +1,560 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Command line tool to interact with Chronicle's Config Based Normalizer APIs.
+
+Config Based Normalizer (CBN) APIs allow customers to manage config based
+parsers that normalize logs published to Chronicle.
+
+This script provides a command line tool to interact with these APIs and manage
+config based parsers. Please see readme file for example usages.
+"""
+
+import argparse
+import base64
+import http
+import json
+import os
+import pathlib
+import sys
+import time
+import urllib
+
+from googleapiclient import _auth
+from google.oauth2 import service_account
+
+AUTHORIZATION_SCOPES = ['https://www.googleapis.com/auth/chronicle-backstory']
+DEFAULT_CREDS_FILE_PATH = os.path.join(os.environ['HOME'],
+                                       '.chronicle_credentials.json')
+
+# URLs for Chronicle CBN API endpoints.
+CHRONICLE_API_V1_URL = 'https://backstory.googleapis.com/v1'
+CHRONICLE_API_EUROPE_V1_URL = 'https://europe-backstory.googleapis.com/v1'
+CHRONICLE_API_ASIA_URL = 'https://asia-southeast1-backstory.googleapis.com/v1'
+CHRONICLE_TEST_API_V1_URL = 'https://test-backstory.sandbox.googleapis.com/v1'
```

Comment:
Can we remove this?

Reply:
I agree, and I also think we should change or get rid of get_connecting_url(), although I don't care if it's done in this PR or not. But just to clarify, do you mean you want the user to pass the URL as a command-line arg instead of a string describing the URL (e.g. get_connecting_url() gets "EUROPE" and returns the European URL), or do you have something else in mind?

Reply:
I think what venkata whats is to remove the test URL for now.

Regarding remiving get_connecting_url let me remove that in another PR.

Reply:
What Krishna said (comment just ended up on block of lines as opposed to one). I wanted to remove TEST_URL. I would like to keep get_connecting_url as it serves a purpose.